libraries/standard/mqtt/src/mqtt.c

```diff
@@ -1954,6 +1954,62 @@ uint16_t MQTT_GetPacketId( MQTTContext_t * pContext )
 
 /*-----------------------------------------------------------*/
 
+MQTTStatus_t MQTT_GetSubAckPayload( const MQTTPacketInfo_t * pSubackPacket,
+                                    uint8_t ** pPayloadStart,
+                                    uint16_t * pPayloadSize )
+{
+    MQTTStatus_t status = MQTTSuccess;
+
+    if( pSubackPacket == NULL )
+    {
+        LogError( ( "Invalid parameter: pSubackPacket is NULL." ) );
+        status = MQTTBadParameter;
+    }
+    else if( pPayloadStart == NULL )
+    {
+        LogError( ( "Invalid parameter: pPayloadStart is NULL." ) );
+        status = MQTTBadParameter;
+    }
+    else if( pPayloadSize == NULL )
+    {
+        LogError( ( "Invalid parameter: pPayloadSize is NULL." ) );
+        status = MQTTBadParameter;
+    }
+    else if( pSubackPacket->type != MQTT_PACKET_TYPE_SUBACK )
+    {
+        LogError( ( "Invalid parameter: Input packet is not a SUBACK packet: "
+                    "ExpectedType=%u, InputType=%u", MQTT_PACKET_TYPE_SUBACK, pSubackPacket->type ) );
```

Comment:
It'd make more sense to log these in hex
```suggestion
                    "ExpectedType=%02x, InputType=%02x", MQTT_PACKET_TYPE_SUBACK, pSubackPacket->type ) );
```